01_financial-well-being/housing-value.qmd

```diff
@@ -0,0 +1,904 @@
+---
+title: "Housing Value"
+author: " Manu Alcalá Kovalski"
+date: now
+format:
+  html:
+    embed-resources: true
+    toc: true
+    toc_float: true
+    code-fold: show
+    code-tools: true
+editor_options: 
+  chunk_output_type: console
+execute:
+  warning: false
+---
+
+```{=html}
+<style>
+@import url('https://fonts.googleapis.com/css?family=Lato&display=swap');
+</style>
+```
+<link rel="stylesheet" href="//fonts.googleapis.com/css?family=Lato" />
+
+This program creates the **Housing Value** metric defined as the ratio between the share of a community’s housing wealth held by a racial or ethnic group to the share of households of the same group metric
+
+Initially written by Jung Hyun Choi and updated by Manu Alcalá Kovalski in 2023-2024.
+
+## Process
+
+-   [Housekeeping](#housekeeping)
+-   [Read Data](#read-data)
+-   [Clean Data](#clean-data)
+-   [Merge Crosswalk](#merge-crosswalk)
+-   [Create Housing Value Metric](#create-metric)
+-   [Evaluate Results](#evaluate-results)
+-   [Export](#export)
+
+## Housekeeping {#sec-housekeeping}
+
+```{r}
+#| label: setup
+
+options(scipen = 999)
+librarian::shelf(tidyverse, ipumsr, here, glue, labelled, skimr, urbnthemes, tidylog, assertr)
+set_urbn_defaults(style = "print")
+```
+
+### Read Data {#sec-read-data}
+
+We extract the 1 year ACS files from the IPUMS API using `ipumsr`. You will need to have an API key in order to do so. If you don't have one, you can register to the [IPUMS USA](https://uma.pop.umn.edu/usa/user/new?return_url=https%3A%2F%2Fusa.ipums.org%2Fusa-action%2Fmenu) data collection and create an API key.
+
+By default, `ipumsr` API functions assume that your key is stored in the **IPUMS_API_KEY** environment variable. You can also provide your key directly to these functions, but storing it in an environment variable saves you some typing and helps prevent you from inadvertently sharing your key with others (for instance, on GitHub).
+
+You can save your API key to the IPUMS_API_KEY environment variable with `set_ipums_api_key(api_key = "paste-your-key-here", save = TRUE)`. This will add your API key to your .Renviron file in your user home directory.
+
+```{r}
+#| label: query-ipums-api
+#| cache: true
+#| eval: false
+## 2.1. ACS: keep households living not living in gq
+
+# Run this the first time you run this code to avoid requesting the same data
+# multiple times
+
+acs_vars <- c("YEAR", "SAMPLE", "SERIAL", "CBSERIAL", "HHWT", "CLUSTER", "state", "county", "MET2013", "CITY", "PUMA", "STRATA", "GQ", "OWNERSHP", "OWNERSHPD", "VALUEH", "PERNUM", "PERWT", "RACE", "RACED", "HISPAN", "HISPAND", "AGE")
+
+years <- 2014:2022
+# usa_ext_def <- define_extract_usa(
+#   description = "ACS 1 Year 2014-2022",
+#   samples = paste0("us", years, "a"),
+#   variables = acs_vars
+# )
+#
+# usa_ext_submitted <- submit_extract(usa_ext_def)
+# usa_ext_complete <- wait_for_extract(usa_ext_submitted)
```

Comment:
Line 75 and beyond is not replicable to me at this time because I get invalid variable name (API request fails with status 400) for state and county. If you look on the IPUMS site, it shows it should be statefip and countyfip. When I change this, I can successfully submit the extract.

Reply:
Changing to statefip and countyfip fixed this. Thank you for flagging!!